### DISCUSSION TITLE
Get the same result every evaluate

### DISCUSSION THREAD

**Discussion by Sakurakdx** [Lightning-AI/pytorch-lightning#10945] [code help: NLP / ASR / TTS]
When I finish training for evaluation, the results obtained in each evaluation are the same, but the parameters in the training process are changed.
When I set the training and evaluation data to be the same and the sample size is 1, the result of every train step is different, but the evaluation result are the same.
My validation_step() is follow:
![image](https://user-images.githubusercontent.com/48399040/144748269-e565d91f-d482-4c5b-8e1e-e4f1ad63f472.png)

**Comment by tchaton** [comment]
Dear @Sakurakdx,

Would you mind reproducing these results with the BoringModel you can find under `pl_examples`?

Furthermore, you might have `shuffle=True` for training data, which might explain the result are different for training.

**Reply by Sakurakdx** [reply]
I set `shuffle=False`.
I load `roberta-large` model, which has more than 400M params. But the lighting shows me my model only have 250M params.
![image](https://user-images.githubusercontent.com/48399040/144861248-ed7876c0-8f76-41db-aaec-939bab5a569c.png)

Is there a problem with the way I initialize the model?
![image](https://user-images.githubusercontent.com/48399040/144861380-80d1b2bb-3e72-4279-bd4d-05478bf2ad3f.png)

**Reply by tchaton** [reply]
It seems you load only the decoder right?

**Reply by rohitgr7** [reply]
are you sure it's roberta-large? standalone roberta-large is 1.8 gigs and it has more params

roberta-base has around 125Mil params though.

**Reply by Sakurakdx** [reply]
sorry, it's my fault. the model is `roberta-base`. And I thought `roberta-base` 480Mil params. Thanks for your correction.

But I want to kown why the model parameters are changed, but the result of each evaluation is still the same.
I check my model, and set the dev data as same as train data. The outputs of train loop will change, but the ouputs of the evaluate loop is stiil the same.

**Reply by Sakurakdx** [reply]
> are you sure it's roberta-large? standalone roberta-large is 1.8 gigs and it has more params
> 
> roberta-base has around 125Mil params though.

Strangely, I didn’t change anything, and ran the code again and again, and the bug disappeared by itself.
Before that, I ran it many times, but the bug persisted. It shocked me a lot.
Anyway, thanks for your respons.

**Reply by rohitgr7** [reply]
glad its working for you now.. ping us if the issue comes up again :)
